app/controllers/client_data_controller.rb

```diff
@@ -94,12 +94,20 @@ def auth_errors(exception)
     path = polymorphic_path(model_class, action: :new)
     # prevent redirect loop
     if path == request.path
-      render "authorization_error", status: 403, locals: { msg: exception.message }
+      render_auth_errors(exception)
     else
       redirect_to path, alert: exception.message
     end
   end
 
+  def render_auth_errors(exception)
+    if exception.message == "Client is disabled"
+      render_disabled_client_message(exception.message)
```

Comment:
it would introduce a regression *if* there weren't a test for the expected behavior. but we do have a test! I see your point that it is coincidental that they return the same response, makes sense to keep as-is.